### PR TITLE
feat(filetype): set ~/.aws files to "conf"

### PR DESCRIPTION
**Pull request by justinmk**
Note: Vim 8.2.5054 added "confini" for this.

todo:

- [x] merge [Vim 8.2.5054](https://github.com/vim/vim/commit/635f48010dcf6d97f3a65b4785e1726ad386d3eb) ?
- [x] wait on https://github.com/vim/vim/pull/11416

### CODE REVIEW

**Comment by clason** [comment]
> merge Vim 8.2.5054 ?

That change was already ported, as far as I can tell. Just needs to be marked as included.

**Comment by justinmk** [comment]
> > merge Vim 8.2.5054 ?
> 
> That change was already ported, as far as I can tell. Just needs to be marked as included.

hmm, I don't see "confini" in `runtime/lua/vim/filetype/detect.lua`

**Comment by zeertzjq** [comment]
It's in `runtime/lua/vim/filetype.lua`

File: runtime/lua/vim/filetype.lua [hunk 1]
```diff
@@ -1792,6 +1792,8 @@ local pattern = {
   ['.*/etc/dnsmasq%.conf'] = 'dnsmasq',
   ['php%.ini%-.*'] = 'dosini',
   ['.*/etc/pacman%.conf'] = 'confini',
+  ['.*/%.aws/config'] = 'confini',
+  ['.*/%.aws/credentials'] = 'confini',
```
[the remark this answers is not in the document]
Comment (zeertzjq) [reply runtime/lua/vim/filetype.lua:1796]: Probably need to `make`?

[the remark this answers is not in the document]
Comment (justinmk) [reply runtime/lua/vim/filetype.lua:1796]: that worked :)

[the remark this answers is not in the document]
Comment (clason) [reply runtime/lua/vim/filetype.lua:1796]: Yes, that file is baked into the binary (unless you disable precompilation, which @bfredl added for exactly such a situation ;))

**Comment by clason** [comment]
I'd prefer if we made these changes as part of the `vim-patch.sh` porting of the upstream change -- that's how we've been doing it, and how I've told contributors to do it. We should stick to our own rules ;)

**Comment by justinmk** [comment]
yes, depends on whether https://github.com/vim/vim/pull/11416 goes in as a runtime patch

**Comment by clason** [comment]
It will not, if experience is any guide :)

**Comment by clason** [comment]
Superseded by #20769

**Pull request closed without merging**